noxfile-template.py

```diff
@@ -183,8 +183,8 @@ def _session_tests(
     session: nox.sessions.Session, post_install: Callable = None
 ) -> None:
     # check for presence of tests
-    test_list = glob.glob("*_test.py") + glob.glob("test_*.py")
-    test_list.extend(glob.glob("tests"))
+    test_list = glob.glob("**/*_test.py", recursive=True) + glob.glob("**/test_*.py", recursive=True)
+    test_list.extend(glob.glob("**/tests", recursive=True))
```

Comment:
I'm worried this might cause a bit too much tests to be invoked that's unrelated (other subdirectories within the product that might not be in scope) since this changes noxfile settings for all of our products in `python-docs-samples`. @parthea do you think this is okay to add?

Reply:
It would be good to check whether unrelated tests are being run to avoid unnecessary use of resources.

Reply:
Don't think this PR picks up on any additional tests. I'll try and test on other products myself

Reply:
Unfortunately this does pick up on additional tests. @m-strzelczyk could you revert these changes?

Reply:
@dandhlee  I've reverted the noxfile changes and adjusted file structure instead.